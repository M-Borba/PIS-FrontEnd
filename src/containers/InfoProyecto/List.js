import React from "react";
import propTypes from "prop-types";
import { Box, Typography } from "@material-ui/core";
import { Chip } from "@mui/material";
import List from "@mui/material/List";

const ListData = ({ title, type, data }) => (
  <>
    <Typography style={{ fontWeight: 600, fontSize: 14 }} variant="body1">
      {title}:
    </Typography>
    <List
      maxWidth={360}
      direction="row"
      sx={{
        width: "100%",
        position: "relative",
        overflow: "auto",
        maxHeight: `${type === "personas" ? 370 : 145}`,
        "& ul": { padding: 0 },
      }}
      subheader={<li />}
    >
      {data?.length !== 0 ? (
        data?.map((item) => {
          return (
            <Chip
              sx={{
                margin: "3px",
                fontFamily: "Nunito Sans",
                background: "#E3E1F3",
              }}
              label={item.full_name}
            />
          );
        })
      ) : (
        <Box style={{ paddingTop: 5 }}>
          <Typography variant="button" display="block" gutterBottom>
            AÃºn no hay {type} asociadas
          </Typography>
        </Box>
      )}
    </List>
  </>
);

ListData.propTypes = {
  title: propTypes.string.isRequired,
  type: propTypes.string.isRequired,
  data: propTypes.object.isRequired,
};

export default ListData;
