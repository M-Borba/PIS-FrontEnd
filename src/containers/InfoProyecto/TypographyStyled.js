import React from "react";
import Typography from "@mui/material/Typography";
import { styled } from "@mui/system";

const TypographyStyled = styled(Typography)(({ theme }) => ({
  fontWeight: 600,
  variant: "body1",
  display: "inline",
  gutterBottom: true,
}));

export default TypographyStyled;
