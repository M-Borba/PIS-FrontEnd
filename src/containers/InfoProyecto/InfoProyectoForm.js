import React from "react";
import Grid from "@mui/material/Grid";
import Divider from "@mui/material/Divider";
import { Box, Typography } from "@material-ui/core";
import FormControl from "@mui/material/FormControl";
import Select from "@mui/material/Select";
import MenuItem from "@mui/material/MenuItem";
import moment from "moment";
import propTypes from "prop-types";

import { renderColor } from "../../utils/utils";
import ListData from "./List";
import CustomButton from "../../components/CustomButton";
import TypographyStyled from "./TypographyStyled";
import { BUTTON_LABELS, PROJECT_LABELS } from "../../config/globalVariables";

const InfoProyectoForm = ({
  handleApplyChanges,
  projectData,
  project_state,
  setProjectState,
  onClose,
  type,
}) => {
  return (
    <form onSubmit={(event) => handleApplyChanges(event, projectData)}>
      <Grid container rowSpacing={1} columnSpacing={{ xs: 1, sm: 2, md: 3 }}>
        <Grid item xs={6}>
          <Box>
            <TypographyStyled>{PROJECT_LABELS.ORGANIZACION}: </TypographyStyled>
            <Typography display="inline" variant="body1">
              {projectData.organization || "-"}
            </Typography>
          </Box>

          <Box mt={2}>
            <TypographyStyled>
              {PROJECT_LABELS.TIPO_PROYECTO}:{" "}
            </TypographyStyled>
            <Typography display="inline" variant="body1">
              {type}
            </Typography>
          </Box>

          <Box mt={2} style={{ display: "flex", gap: "0.5rem" }}>
            <TypographyStyled style={{ alignSelf: "center" }}>
              {PROJECT_LABELS.ESTADO}:{" "}
            </TypographyStyled>
            <FormControl>
              <Select
                labelId="state-input"
                label={PROJECT_LABELS.ESTADO}
                id="project_state"
                name="project_state"
                style={{ width: "unset" }}
                value={project_state}
                onChange={(e) => {
                  setProjectState(e.target.value);
                }}
              >
                <MenuItem value="verde">{renderColor("verde")}</MenuItem>
                <MenuItem value="amarillo">{renderColor("amarillo")}</MenuItem>
                <MenuItem value="rojo">{renderColor("rojo")}</MenuItem>
                <MenuItem value="upcoming">{renderColor("upcoming")}</MenuItem>
              </Select>
            </FormControl>
          </Box>

          <Box mt={2}>
            <TypographyStyled>{PROJECT_LABELS.FECHA_INICIO}: </TypographyStyled>
            <Typography display="inline" variant="body1">
              {moment(projectData.start_date).format("DD/MM/YYYY")}
            </Typography>
          </Box>
          <Box mt={2}>
            <TypographyStyled>{PROJECT_LABELS.FECHA_FIN}: </TypographyStyled>
            <Typography display="inline" variant="body1">
              {projectData.end_date
                ? moment(projectData.end_date).format("DD/MM/YYYY")
                : PROJECT_LABELS.INDEFINIDA}
            </Typography>
          </Box>

          <Box mt={2}>
            <TypographyStyled>Descripción: </TypographyStyled>
            <Typography display="inline" variant="body1">
              {projectData.description}
            </Typography>
          </Box>

          {projectData.budget && (
            <Box mt={2}>
              <TypographyStyled>{PROJECT_LABELS.BUDGET}: </TypographyStyled>
              <Typography display="inline" variant="body1">
                {projectData.budget}
              </Typography>
            </Box>
          )}
        </Grid>

        <Grid
          item
          style={{ display: "flex", padding: 0, flexDirection: "column" }}
          xs={6}
        >
          <Box>
            <ListData
              title="Tecnologías"
              type="tecnologías"
              data={projectData.technologies}
            />
          </Box>
          <Divider style={{ marginBottom: 15, marginTop: 15 }} />
          <Box>
            <ListData
              title="Personas Asignadas"
              type="personas"
              data={projectData.people}
            />
          </Box>
          <Grid
            xs={12}
            style={{
              display: "flex",
              flexDirection: "row",
              width: "100%",
            }}
          >
            <Grid
              xs={6}
              style={{
                display: "flex",
                flexDirection: "row",
                maxHeight: "45px",
                justifyContent: "flex-start",
                width: "100%",
                alignSelf: "flex-end",
              }}
            >
              <CustomButton
                styles={{
                  backgroundColor: "#c21321",
                  maxHeight: "45px",
                  "&:hover": { backgroundColor: "#9d1e29" },
                }}
                variant="contained"
                onClick={onClose}
              >
                {BUTTON_LABELS.CANCEL}
              </CustomButton>
            </Grid>
            <Grid
              xs={12}
              style={{
                display: "flex",
                flexDirection: "row",
                justifyContent: "flex-end",
                width: "100%",
                alignSelf: "flex-end",
              }}
            >
              <CustomButton type="submit" variant="contained">
                {BUTTON_LABELS.APPLY_CHANGES}
              </CustomButton>
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </form>
  );
};

InfoProyectoForm.propTypes = {
  projectData: propTypes.object.isRequired,
  onClose: propTypes.func.isRequired,
  handleApplyChanges: propTypes.func.isRequired,
  project_state: propTypes.string.isRequired,
  setProjectState: propTypes.func.isRequired,
  type: propTypes.string.isRequired,
};

export default InfoProyectoForm;
