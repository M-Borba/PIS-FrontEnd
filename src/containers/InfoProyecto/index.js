import React, { useState } from "react";
import propTypes from "prop-types";
import { Box, Typography } from "@material-ui/core";
import Grid from "@mui/material/Grid";
import Divider from "@mui/material/Divider";
import AssignmentIcon from "@mui/icons-material/Assignment";
import { useSnackbar } from "notistack";
import FormControl from "@mui/material/FormControl";
import Select from "@mui/material/Select";
import MenuItem from "@mui/material/MenuItem";
import moment from "moment";

import ListData from "./List";
import AddButton from "../../components/AddButton";

import { renderColor } from "../../utils/utils";
import { axiosInstance } from "../../config/axios";
import {
  BUTTON_LABELS,
  PROYECT_FORM_LABELS,
} from "../../config/globalVariables";
import TypographyStyled from "./TypographyStyled";

InfoProyecto.propTypes = {
  projectData: propTypes.object.isRequired,
  type: propTypes.string,
  onClose: propTypes.func,
};

export default function InfoProyecto({ projectData, type, onClose }) {
  const [project_state, setProjectState] = useState(projectData.project_state);
  const initialState = projectData.project_state;
  const { enqueueSnackbar } = useSnackbar();

  const handleApplyChanges = (e, projectData) => {
    if (initialState !== project_state) {
      projectData.project_state = project_state;
      axiosInstance
        .put(`/projects/${projectData.id}`, {
          project: projectData,
        })
        .then((response) => {
          enqueueSnackbar(
            `El proyecto ${projectData.name} se actualizó con éxito.`,
            {
              variant: "success",
              autoHideDuration: 4000,
            }
          );
        })
        .catch((error) => {
          enqueueSnackbar(
            `El proyecto ${projectData.name} no se actualizó con éxito.`,
            {
              variant: "error",
              autoHideDuration: 4000,
            }
          );
        });
    } else {
      e.preventDefault();
      onClose();
    }
  };

  return (
    <div style={{ padding: 30 }}>
      <div
        style={{
          display: "flex",
          alignItems: "center",
          flexWrap: "wrap",
        }}
      >
        <AssignmentIcon
          style={{
            width: 50,
            height: 50,
          }}
        />
        <Typography
          variant="h4"
          style={{ marginLeft: 10, overflowWrap: "break-word" }}
        >
          {projectData.name}
        </Typography>
      </div>
      <Divider style={{ marginBottom: 15, marginTop: 15 }} />
      <form onSubmit={(event) => handleApplyChanges(event, projectData)}>
        <Grid container rowSpacing={1} columnSpacing={{ xs: 1, sm: 2, md: 3 }}>
          <Grid item xs={6}>
            <Box>
              <TypographyStyled>
                {PROYECT_FORM_LABELS.ORGANIZACION}:{" "}
              </TypographyStyled>
              <Typography display="inline" variant="body1">
                {projectData.organization || "-"}
              </Typography>
            </Box>

            <Box mt={2}>
              <TypographyStyled>
                {PROYECT_FORM_LABELS.TIPO_PROYECTO}:{" "}
              </TypographyStyled>
              <Typography display="inline" variant="body1">
                {type}
              </Typography>
            </Box>

            <Box mt={2} style={{ display: "flex", gap: "0.5rem" }}>
              <TypographyStyled style={{ alignSelf: "center" }}>
                {PROYECT_FORM_LABELS.ESTADO}:{" "}
              </TypographyStyled>
              <FormControl>
                <Select
                  labelId="state-input"
                  label="Estado"
                  id="project_state"
                  name="project_state"
                  style={{ width: "unset" }}
                  value={project_state}
                  onChange={(e) => {
                    setProjectState(e.target.value);
                  }}
                >
                  <MenuItem value="verde">{renderColor("verde")}</MenuItem>
                  <MenuItem value="amarillo">
                    {renderColor("amarillo")}
                  </MenuItem>
                  <MenuItem value="rojo">{renderColor("rojo")}</MenuItem>
                  <MenuItem value="upcoming">
                    {renderColor("upcoming")}
                  </MenuItem>
                </Select>
              </FormControl>
            </Box>

            <Box mt={2}>
              <TypographyStyled>
                {PROYECT_FORM_LABELS.FECHA_INICIO}:{" "}
              </TypographyStyled>
              <Typography display="inline" variant="body1">
                {moment(projectData.start_date).format("DD/MM/YYYY")}
              </Typography>
            </Box>
            <Box mt={2}>
              <TypographyStyled>
                {PROYECT_FORM_LABELS.FECHA_FIN}:{" "}
              </TypographyStyled>
              <Typography display="inline" variant="body1">
                {projectData.end_date
                  ? moment(projectData.end_date).format("DD/MM/YYYY")
                  : PROYECT_FORM_LABELS.INDEFINIDA}
              </Typography>
            </Box>

            <Box mt={2}>
              <TypographyStyled>Descripción: </TypographyStyled>
              <Typography display="inline" variant="body1">
                {projectData.description}
              </Typography>
            </Box>

            {projectData.budget && (
              <Box mt={2}>
                <TypographyStyled>
                  {PROYECT_FORM_LABELS.BUDGET}:{" "}
                </TypographyStyled>
                <Typography display="inline" variant="body1">
                  {projectData.budget}
                </Typography>
              </Box>
            )}
          </Grid>

          <Grid
            item
            style={{ display: "flex", padding: 0, flexDirection: "column" }}
            xs={6}
          >
            <Box>
              <ListData
                title="Tecnologías"
                type="tecnologías"
                data={projectData.technologies}
              />
            </Box>
            <Divider style={{ marginBottom: 15, marginTop: 15 }} />
            <Box>
              <ListData
                title="Personas Asignadas"
                type="personas"
                data={projectData.people}
              />
            </Box>
            <Grid
              xs={12}
              style={{
                display: "flex",
                flexDirection: "row",
                width: "100%",
              }}
            >
              <Grid
                xs={12}
                style={{
                  display: "flex",
                  flexDirection: "row",
                  maxHeight: "45px",
                  justifyContent: "flex-start",
                  width: "100%",
                  alignSelf: "flex-end",
                }}
              >
                <AddButton type="submit" variant="contained">
                  {BUTTON_LABELS.APPLY_CHANGES}
                </AddButton>
              </Grid>
              <Grid
                xs={6}
                style={{
                  display: "flex",
                  flexDirection: "row",
                  justifyContent: "flex-end",
                  width: "100%",
                  alignSelf: "flex-end",
                }}
              >
                <AddButton
                  styles={{
                    backgroundColor: "#c21321",
                    maxHeight: "45px",
                    "&:hover": { backgroundColor: "#9d1e29" },
                  }}
                  variant="contained"
                  onClick={onClose}
                >
                  {BUTTON_LABELS.CANCEL}
                </AddButton>
              </Grid>
            </Grid>
          </Grid>
        </Grid>
      </form>
    </div>
  );
}
