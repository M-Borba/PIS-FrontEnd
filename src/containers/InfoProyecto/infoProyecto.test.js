import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import ReactDOM from "react-dom";
import InfoProyecto from "./index";
import React from "react";

describe("project list component tests", () => {
  it("renders without crashing", () => {
    const pData = {
      name: "Nombre",
      description: "Descripcion",
      budget: 400,
    }; //actualizar en el futuro con lista de personas y tecnologias
    const div = document.createElement("div");
    ReactDOM.render(<InfoProyecto projectData={pData} />, div);
  });
  it("renders correct data", () => {
    const pData = {
      name: "Nombre",
      description: "Descripcion",
      budget: 400,
    }; //actualizar en el futuro con lista de personas y tecnologias
    const { container } = render(<InfoProyecto projectData={pData} />);
    // Matches the last snapshot taken
    expect(container).toMatchSnapshot();
  });
});
