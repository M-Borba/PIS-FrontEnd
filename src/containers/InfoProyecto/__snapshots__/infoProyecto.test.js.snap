// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`project list component tests renders correct data 1`] = `
<div>
  <div
    style="padding: 16px;"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-[object Object] css-1hxutbw-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-hdppi2-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-h5 MuiTypography-paragraph"
          style="overflow-wrap: break-word;"
        >
          Nombre
        </p>
        <div
          class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
          style="margin-bottom: 10px; max-height: 397px; overflow: auto;"
        >
          <div
            class="MuiCardContent-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
            >
              Descripcion
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
            >
              Descripcion
            </p>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="MuiCardContent-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
            >
              Presupuesto
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
            >
              400
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1744jwa-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="MuiCardContent-root"
          >
            <ul
              class="MuiList-root MuiList-padding MuiList-subheader css-1x2ob93-MuiList-root"
            >
              <li />
              <li>
                <ul>
                  <li
                    class="MuiListSubheader-root MuiListSubheader-gutters MuiListSubheader-sticky css-uob957-MuiListSubheader-root"
                  >
                    Personas Asignadas
                  </li>
                  <li
                    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                  >
                    <div
                      class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                      >
                        Juan con nombre muy muy muy muy muy largo
                      </span>
                    </div>
                  </li>
                  <li
                    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                  >
                    <div
                      class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                      >
                        Juan2
                      </span>
                    </div>
                  </li>
                  <li
                    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                  >
                    <div
                      class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                      >
                        Juan3
                      </span>
                    </div>
                  </li>
                  <li
                    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                  >
                    <div
                      class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                      >
                        Juan4
                      </span>
                    </div>
                  </li>
                  <li
                    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                  >
                    <div
                      class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                      >
                        Juan5
                      </span>
                    </div>
                  </li>
                  <li
                    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                  >
                    <div
                      class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                      >
                        Juan6
                      </span>
                    </div>
                  </li>
                  <li
                    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                  >
                    <div
                      class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                      >
                        Juan7
                      </span>
                    </div>
                  </li>
                  <li
                    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                  >
                    <div
                      class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                      >
                        Juan7
                      </span>
                    </div>
                  </li>
                  <li
                    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                  >
                    <div
                      class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                      >
                        Juan6
                      </span>
                    </div>
                  </li>
                  <li
                    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                  >
                    <div
                      class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                      >
                        Juan7
                      </span>
                    </div>
                  </li>
                  <li
                    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                  >
                    <div
                      class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                      >
                        Juan7
                      </span>
                    </div>
                  </li>
                </ul>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-j23p4t-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="MuiCardContent-root"
          >
            <ul
              class="MuiList-root MuiList-padding MuiList-subheader css-1x2ob93-MuiList-root"
            >
              <li />
              <li>
                <ul>
                  <li
                    class="MuiListSubheader-root MuiListSubheader-gutters MuiListSubheader-sticky css-uob957-MuiListSubheader-root"
                  >
                    Tecnolog√≠as
                  </li>
                  <li
                    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                  >
                    <div
                      class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                      >
                        Ruby on Rails v3.1
                      </span>
                    </div>
                  </li>
                  <li
                    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                  >
                    <div
                      class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                      >
                        Java2
                      </span>
                    </div>
                  </li>
                  <li
                    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                  >
                    <div
                      class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                      >
                        Java3
                      </span>
                    </div>
                  </li>
                  <li
                    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                  >
                    <div
                      class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                      >
                        Java4
                      </span>
                    </div>
                  </li>
                  <li
                    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                  >
                    <div
                      class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                      >
                        Java5
                      </span>
                    </div>
                  </li>
                  <li
                    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                  >
                    <div
                      class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                      >
                        Java6
                      </span>
                    </div>
                  </li>
                </ul>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
